packages/amplication-client/src/Resource/break-the-monolith/BtmButton.tsx

```diff
@@ -120,39 +172,62 @@ export const BtmButton: React.FC<Props> = ({
+          element={
+            <Button
+              iconPosition={EnumIconPosition.Left}
+              buttonStyle={ButtonStyle}
+              onClick={onButtonSelectResource}
+              disabled={!hasRedesignArchitectureFeature}
+              icon={"ai"}
+            >
+              {buttonText}
+            </Button>
+          }
+        ></FeatureIndicator>
       ) : (
-        <SelectMenu
-          title={buttonText}
-          buttonStyle={ButtonStyle}
-          hideSelectedItemsIndication
-        >
-          <SelectMenuModal align="right" withCaret>
-            <SelectMenuList>
-              {resources.map((resource) => (
-                <SelectMenuItem
-                  key={resource.id}
-                  closeAfterSelectionChange
-                  itemData={resource}
-                  onSelectionChange={onSelectMenuSelectResource}
-                >
-                  <FlexItem
-                    itemsAlign={EnumItemsAlign.Center}
-                    end={<Icon icon={"app-settings"} size="xsmall"></Icon>}
-                  >
-                    <ResourceCircleBadge
-                      type={resource.resourceType}
-                      size="small"
-                    />
-                    <span>{resource.name}</span>
-                  </FlexItem>
-                </SelectMenuItem>
-              ))}
-            </SelectMenuList>
-          </SelectMenuModal>
-        </SelectMenu>
+        <FeatureIndicator
+          featureName={BillingFeature.RedesignArchitecture}
+          text={tooltipText}
+          linkText={upgradeTooltipText}
+          element={
+            <SelectMenu
+              title={buttonText}
+              buttonStyle={ButtonStyle}
+              hideSelectedItemsIndication
+              disabled={!hasRedesignArchitectureFeature}
+              icon={"ai"}
+              buttonIconPosition={EnumIconPosition.Left}
+            >
+              <SelectMenuModal align="right" withCaret>
+                <SelectMenuList>
+                  {resources.map((resource) => (
+                    <SelectMenuItem
+                      key={resource.id}
+                      closeAfterSelectionChange
+                      itemData={resource}
+                      onSelectionChange={onSelectMenuSelectResource}
+                    >
+                      <FlexItem
+                        itemsAlign={EnumItemsAlign.Center}
+                        end={<Icon icon={"app-settings"} size="xsmall"></Icon>}
+                      >
+                        <ResourceCircleBadge
+                          type={resource.resourceType}
+                          size="small"
+                        />
+                        <span>{resource.name}</span>
+                      </FlexItem>
+                    </SelectMenuItem>
+                  ))}
+                </SelectMenuList>
+              </SelectMenuModal>
+            </SelectMenu>
+          }
```

Comment:
Aren't you missing a `</FeatureIndicatore>` here?

Reply:
No. it can also be used as a self-closing tag. If it was mandatory, the CI would fail.
I updated the other one to be self-closing as well. `FeatureIndicator` doesn't have children, so it's just confusing like that.